tests/oc_template_test.go

```diff
@@ -0,0 +1,328 @@
+											"disk": {
+												"bus": "virtio"
+											},
+											"name": "disk0"
+										},
+										{
+											"volumeName": "cloudinitvolume",
+											"disk": {
+												"bus": "virtio"
+											},
+											"name": "disk1"
+										}
+									]
+								}
+							},
+							"volumes": [
+								{
+									"name": "registryvolume",
+									"registryDisk": {
+										"image": "kubevirt/fedora-cloud-registry-disk-demo:latest"
+									}
+								},
+								{
+									"name": "cloudinitvolume",
+									"cloudInitNoCloud": {
+										"userDataBase64": "I2Nsb3VkLWNvbmZpZwpwYXNzd29yZDogYXRvbWljCnNzaF9wd2F1dGg6IFRydWUKY2hwYXNzd2Q6IHsgZXhwaXJlOiBGYWxzZSB9Cg=="
+									}
+								}
+							]
+						},
+						"metadata": {
+							"labels": {
+								"kubevirt-ovm": "ovm-${NAME}"
+							}
+						}
+					}
+				},
+				"apiVersion": "kubevirt.io/v1alpha1",
+				"metadata": {
+					"labels": {
+						"kubevirt-ovm": "ovm-${NAME}"
+					},
+					"name": "${NAME}"
+				}
+			}
+		],
+		"apiVersion": "v1",
+		"parameters": [
+			{
+				"name": "NAME",
+				"description": "Name for the new VM"
+			},
+			{
+				"name": "MEMORY",
+				"value": "4096Mi",
+				"description": "Amount of memory"
+			},
+			{
+				"name": "CPU_CORES",
+				"value": "4",
+				"description": "Amount of cores"
+			}
+		],
+		"metadata": {
+			"labels": {
+				"miq.github.io/kubevirt-is-vm-template": "true",
+				"kubevirt.io/os": "fedora27"
+			},
+			"annotations": {
+				"description": "OCP KubeVirt Fedora 27 VM template",
+				"tags": "kubevirt,ocp,template,linux"
+			},
+			"console.openshift.io/iconClass": "icon-fedora",
+			"name": "fedora-vm-template"
+		}
+	}`
+
+	Context("with oc command", func() {
+		var vmTemplateJsonFile = ""
+		var vmJsonFile = ""
+
+		BeforeEach(func() {
+			tests.SkipIfNoOc()
+			// write testing vm template into file
+			vmTemplateJsonFile, err = WriteJson("testvm-template", vmTemplate)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			if vmTemplateJsonFile != "" {
+				// remove testing vm template
+				err = os.Remove(vmTemplateJsonFile)
+				Expect(err).ToNot(HaveOccurred())
+				vmTemplateJsonFile = ""
+			}
+			if vmJsonFile != "" {
+				// remove testing vm JSON file
+				err = os.Remove(vmJsonFile)
+				Expect(err).ToNot(HaveOccurred())
+				vmJsonFile = ""
+			}
+		})
+
+		It("should convert an offline vm template into JSON file", func() {
+			By("Converting offline vm template into JSON file via oc-process command")
+			out, err = tests.RunOcCommand("process", "-f", vmTemplateJsonFile, vmTemplateParams)
+			Expect(err).ToNot(HaveOccurred())
+			vmJsonFile, err = WriteJson(vmName, out)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should create an offline vm from a template", func() {
+			var message = ""
+			// probably there is a better way how to do this without duplicating the code
+			By("Converting offline vm template into JSON file via oc-process command")
```

Comment:
I think you can move this section to separate function under `Describe`
```
generateVmFromTemplate := func (vmName string) {
    By("Converting offline vm template into JSON file via oc-process command")
    out, err = tests.RunOcCommand("process", "-f", vmTemplateJsonFile, vmTemplateParams)
    Expect(err).ToNot(HaveOccurred())
    vmJsonFile, err = WriteJson(vmName, out)
    Expect(err).ToNot(HaveOccurred())
}
```
the same true for start and stop VM via `oc` command

Reply:
@cynepco3hahue I like this idea more than the `table.DescribeTable`. Thanks for the suggestion.